net/NetUtil.cpp

```diff
@@ -140,24 +140,28 @@ static HostEntry resolveDNS(const std::string& addressToCheck,
 {
     const auto now = std::chrono::steady_clock::now();
 
-    // remove entries >= 20 seconds old
-    std::erase_if(querycache, [now](const auto& entry)->bool {
```

Comment:
It's true that this would remove _all_ expired references, not the one we lookup.

This means that entries that are stale, will remain in the cache unless/until there is a lookup, accumulating entries.

I suspect this is highly unlikely, but possible. As a workaround, we can revise this to do `erase_if` when we have an expired entry. That is, the first entry that is expired, also prompts us to erase any _other_ entries that might be expired as well. This way, we don't get the double-lookup hit on each lookup.